src/analytics/lambdas/scan-metadata-workflow/check-metadata-workflow-start.ts

```diff
@@ -0,0 +1,188 @@
+    jobStartTimestamp: currentTimestamp,
+    scanStartDate: startScanDate,
+  };
+ */
+export const handler = async (event: CheckMetadataWorkflowEvent) => {
+  logger.debug('request event:', JSON.stringify(event));
+  try {
+    const eventSource = event.eventSource;
+    // workflow is triggered from upstream workflow
+    if (eventSource === 'LoadDataFlow') {
+      return await handleEventFromUpstreamWorkflow();
+    } else {
+      const inputScanEndDate = formatScanDate(event.scanEndDate);
+      const inputScanStartDate = formatScanDate(event.scanStartDate);
+      const currentTimestamp = Date.now();
+      const scanEndDate = getScanEndDate(inputScanEndDate, getDateFromTimestamp(currentTimestamp));
+      let scanStartDate = inputScanStartDate;
+      return {
+        status: WorkflowStatus.CONTINUE,
+        scanEndDate: scanEndDate,
+        jobStartTimestamp: currentTimestamp,
+        scanStartDate: scanStartDate,
+      };
+    }
+  } catch (err) {
+    if (err instanceof Error) {
+      logger.error('Error when scan metadata.', err);
+    }
+    throw err;
+  }
+};
+
+async function handleEventFromUpstreamWorkflow() {
+  const currentTimestamp = Date.now();
+  const scanEndDate = await getMaxScanEndDate();
+  const response = await getWorkflowInfoFromS3();
+  let result = {};
+  if (response) {
+    const lastJobStartTimestamp = response.lastJobStartTimestamp;
+    const startScanDate = response.lastScanEndDate;
+
+    // Triggered if more than 24 hours have passed since the last job execution
+    const workflowMinInterval = parseInt(process.env.WORKFLOW_MIN_INTERVAL || '1440');
+    if (!lastJobStartTimestamp || currentTimestamp - lastJobStartTimestamp >= workflowMinInterval * 60 * 1000) {
+      result = {
+        status: WorkflowStatus.CONTINUE,
+        scanEndDate: scanEndDate,
+        jobStartTimestamp: currentTimestamp,
+        scanStartDate: startScanDate,
+      };
+    } else {
+      result = {
+        status: WorkflowStatus.SKIP,
+      };
+    }
+  } else {
+    // first time to execution, no job info in ddb.
+    result = {
+      status: WorkflowStatus.CONTINUE,
+      scanEndDate: scanEndDate,
+      jobStartTimestamp: currentTimestamp,
+      scanStartDate: '',
+    };
+  }
+  return result;
+}
+
+async function getWorkflowInfoFromS3() {
+  try {
+    const s3Key = getWorkflowInfoKey(pipelineS3Prefix, projectId);
+
+    return await readS3ObjectAsJson(
+      pipelineS3BucketName,
+      s3Key,
+    );
+  } catch (error) {
+    if (error instanceof Error) {
+      logger.error('Error get workflow info data from s3:', error);
+    }
+    throw error;
+  }
+}
+
+// get latest date from emr ods job file
+async function getMaxScanEndDate() {
+  try {
+    const latestJobTimestamp = await getLatestEmrJobEndTime(pipelineS3BucketName, pipelineS3Prefix, projectId);
+    let maxScanEndDate = undefined;
+    if (latestJobTimestamp) {
+      logger.info(
+        'found emr latest job timestamp',
```

Comment:
It'd better logging `latestJobTimestamp` value too.

Reply:
Done